kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/util.kt

```diff
@@ -234,3 +214,22 @@ internal fun ClassId.toKtClassSymbol(): KtClassOrObjectSymbol? {
         }
     }
 }
+
+internal fun KtType.classifierSymbol(): KtClassifierSymbol? {
+    return try {
+        when (this) {
+            is KtTypeParameterType -> this.symbol
+            is KtCapturedType -> TODO("fix in upstream")
+            is KtClassErrorType -> null
+            is KtFunctionalType -> classSymbol
+            is KtUsualClassType -> classSymbol
+            is KtDefinitelyNotNullType -> original.classifierSymbol()
+            is KtDynamicType -> null
+            is KtFlexibleType -> lowerBound.classifierSymbol()
+            is KtIntegerLiteralType -> null
+            is KtIntersectionType -> null
```

Comment:
This is probably overly simplified for `KtFlexibleType` and `KtIntersectionType`, especially for the latter. Can you comment why returning `lowerBound` for the flexible, and make the intersection `TODO()`? Or, comment this function to make it explicit that some types are not supported / not applicable.